hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -459,6 +459,7 @@ private OzoneManager(OzoneConfiguration conf, StartupOption startupOption)
     this.omNodeDetails = omhaNodeDetails.getLocalNodeDetails();
 
     omStorage = new OMStorage(conf);
+    omStorage.validateOrPersistOmNodeId(omNodeDetails.getNodeId());
```

Comment:
Comment on the sidenote:
As a possible solution we can bump the layout version, and add an UpgradeAction to handle the change in the VERSION file, however I am not sure whether this is an overkill for just adding this value. Probably it is not, as we are changing the metadata, but I am still a bit hesitant to go and do it without hearing your opinion on this :)

I thought adding a comment here is just redundant, as the doc of the method is pretty much trying to tell what it is doing, but anyway... why not :) I have added a code comment before the method call to explain, however if we decide to do this within the upgrade framework, this one will be meaningless anyway ;)

Reply:
I can see 3 possible options:

1. Write the value in the constructor if it is not present, or validate the value if it is already there.
    - Simple approach
    - All installs get the value without a re-init.
    - May lead to code fragmentation between omInit and the constructor for all new fields added to the version file.
        - omInit must write all the values, but the constructor must also write any new values. Previously the constructor did not write to the version file.
2. Add a layout feature and upgrade action to write the new value to the version file. omInit writes the value to the version file, and OM constructor only validates the value.
    - More involved approach
    - All installs get the value without a re-init.
    - omInit and the constructor maintain their original jobs with respect to the version file, but some upgrade pieces must be added.
3. Write the value in omInit only. Existing clusters that do not re-init will not have the value persisted.
    - Simple approach
    - Only new installs or re-inited existing installs get it.
    - omInit and constructor maintain their original jobs with respect to the version file.

2 seems excessive IMO since there are no downgrade implications to the change. 3 may be annoying to debug in the future. Someone may change the node ID and this change will not catch it even if it is present. So after writing out the options like this, I think 1 is the best choice, which is what the PR is already doing.

After another look the javadoc on the method should be sufficient. If anyone is wondering why the constructor may need to modify the version file on startup they would probably go from the constructor to the method declaration and see the doc.

Reply:
Ok, I think we agree on this then. I have added these options as a comment above the validateOrStore method.

Reply:
Oh, one more thing...
I went ahead and tried to add an UpgradeAction, but once I started I quickly realized the following:
UpgradeActions are running pretty much after Ozone Manager was initialized almost fully, in the start() method, it seems to be pretty late to fail, and the impacts of moving the initialization of the value in the storage or the validation to that late are not clear, so it is at least risky.